client/dashboard/me/preferences-new-hosting-dashboard/index.tsx

```diff
@@ -0,0 +1,141 @@
+import { userPreferenceQuery, userPreferenceMutation } from '@automattic/api-queries';
+import { Badge } from '@automattic/ui';
+import { useMutation, useSuspenseQuery } from '@tanstack/react-query';
+import {
+	Button,
+	Card,
+	CardBody,
+	CheckboxControl,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { DataForm } from '@wordpress/dataviews';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { Text } from '../../components/text';
+import type { Field } from '@wordpress/dataviews';
+
+interface OptInFormData {
+	enabled: boolean;
+}
+
+const form = {
+	layout: {
+		type: 'regular' as const,
+		labelPosition: 'top' as const,
+	},
+	fields: [
+		{
+			id: 'optInForm',
+			label: (
+				<HStack spacing={ 1 }>
+					<span>{ __( 'Try the new Hosting Dashboard' ) }</span>
+					<Badge intent="info">{ __( 'New feature' ) }</Badge>
+				</HStack>
```

Comment:
I'm surprised this works tbh. The TypeScript says that `label` is only allowed to be a `string`.